test/gtest/ucp/test_ucp_wfe.cc

```diff
@@ -0,0 +1,102 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2014.  ALL RIGHTS RESERVED.
+*
+* Copyright (C) UT-Battelle, LLC. 2015. ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+#include "ucp_test.h"
+
+#include <gtest/common/test_perf.h>
+
+
+#define MB   pow(1024.0, -2)
+#define UCP_ARM_PERF_TEST_MULTIPLIER 2
+class test_ucp_wfe : public ucp_test, public test_perf {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant>& variants) {
+        add_variant(variants, 0);
+    }
+
+protected:
+    virtual void init() {
+        test_base::init(); /* Skip entities creation in ucp_test */
+        ucs_log_push_handler(log_handler);
+    }
+
+    virtual void cleanup() {
+        ucs_log_pop_handler();
+        test_base::cleanup();
+    }
+
+    static ucs_log_func_rc_t
+    log_handler(const char *file, unsigned line, const char *function,
+                ucs_log_level_t level,
+                const ucs_log_component_config_t *comp_conf,
+                const char *message, va_list ap) {
+        // Ignore errors that transport cannot reach peer
+        if (level == UCS_LOG_LEVEL_ERROR) {
+            std::string err_str = format_message(message, ap);
+            if ((err_str.find(ucs_status_string(UCS_ERR_UNREACHABLE)) != std::string::npos) ||
+                (err_str.find(ucs_status_string(UCS_ERR_UNSUPPORTED)) != std::string::npos)) {
+                UCS_TEST_MESSAGE << err_str;
+                return UCS_LOG_FUNC_RC_STOP;
+            }
+        }
+        return UCS_LOG_FUNC_RC_CONTINUE;
+    }
+
+    const static test_spec tests[];
+};
+
+
+const test_perf::test_spec test_ucp_wfe::tests[] =
+{
+    { "put latency", "usec",
+      UCX_PERF_API_UCP, UCX_PERF_CMD_PUT, UCX_PERF_TEST_TYPE_PINGPONG,
+      UCP_PERF_DATATYPE_CONTIG, 0, 1, { 8 }, 1, 1000lu,
+      ucs_offsetof(ucx_perf_result_t, latency.total_average), 1e6, 0.001, 30.0,
+      0 },
+
+    { "put latency with WFE", "usec",
+      UCX_PERF_API_UCP, UCX_PERF_CMD_PUT, UCX_PERF_TEST_TYPE_PINGPONG_WFE,
+      UCP_PERF_DATATYPE_CONTIG, 0, 1, { 8 }, 1, 1000lu,
+      ucs_offsetof(ucx_perf_result_t, latency.total_average), 1e6, 0.001, 30.0,
+      0 }
+};
+
+
+UCS_TEST_P(test_ucp_wfe, envelope) {
+    double perf_value = 0;
+    int cache_perf_retry_count;
+    test_spec test;
+
+    std::stringstream ss;
+    ss << GetParam();
+    /* coverity[tainted_string_argument] */
+    ucs::scoped_setenv tls("UCX_TLS", "shm");
```

Comment:
it's not a good way
better to use
```
UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_wfe, shm, "shm,")
```
instead of
```
UCP_INSTANTIATE_TEST_CASE(test_ucp_wfe)
```

Reply:
@dmitrygx It is single use case intentionally. The test check that WFE implementation does not introduce substantial latency overheads. First we are running the test without WFE and then follow with test with WFE. Then we compare difference between the two runs and signal failure if it is above the threshold.

Reply:
I will change to UCP_INSTANTIATE_TEST_CASE_TLS

Reply:
> @dmitrygx It is single use case intentionally. The test check that WFE implementation does not introduce substantial latency overheads. First we are running the test without WFE and then follow with test with WFE. Then we compare difference between the two runs and signal failure if it is above the threshold.

I guess it is reply to https://github.com/openucx/ucx/pull/5861#discussion_r520691897
I see the idea, thank you.
Maybe define enum for `0` and `1` that are used to get a perf test suite

Reply:
Fixed 2d1d360